tests/src/performance/Scenario/JitBench/JitBenchHarness.cs

```diff
@@ -0,0 +1,489 @@
+            using (var client = new HttpClient())
+            {
+                var archiveName = $"{JitBenchCommitSha1Id}.zip";
+                var url = $"{JitBenchRepoUrl}/archive/{archiveName}";
+                var zipFile = Path.Combine(s_temporaryDirectory, archiveName);
+
+                using (FileStream tmpzip = File.Create(zipFile))
+                {
+                    using (Stream stream = client.GetStreamAsync(url).Result)
+                        stream.CopyTo(tmpzip);
+                    tmpzip.Flush();
+                }
+
+                // If the repo already exists, we delete it and extract it again.
+                if (Directory.Exists(s_jitBenchDevDirectory))
+                    Directory.Delete(s_jitBenchDevDirectory, true);
+
+                // This step will create s_JitBenchDevDirectory.
+                ZipFile.ExtractToDirectory(zipFile, s_temporaryDirectory);
+                Directory.Move(Path.Combine(s_temporaryDirectory, $"JitBench-{JitBenchCommitSha1Id}"), s_jitBenchDevDirectory);
+            }
+        }
+
+        private static void InstallSharedRuntime()
+        {
+            var psi = new ProcessStartInfo() {
+                WorkingDirectory = s_jitBenchDevDirectory,
+                FileName = @"powershell.exe",
+                Arguments = $".\\Dotnet-Install.ps1 -SharedRuntime -InstallDir .dotnet -Channel master -Architecture {s_targetArchitecture}"
+            };
+            LaunchProcess(psi, 180000);
+        }
+
+        private static IDictionary<string, string> InstallDotnet()
+        {
+            var psi = new ProcessStartInfo() {
+                WorkingDirectory = s_jitBenchDevDirectory,
+                FileName = @"powershell.exe",
+                Arguments = $".\\Dotnet-Install.ps1 -InstallDir .dotnet -Channel master -Architecture {s_targetArchitecture}"
+            };
+            LaunchProcess(psi, 180000);
+
+            // TODO: This is currently hardcoded, but we could probably pull it from the powershell cmdlet call.
+            return new Dictionary<string, string> { { "PATH", $"{Path.Combine(s_jitBenchDevDirectory, ".dotnet")};{psi.Environment["PATH"]}" } };
+        }
+
+        private static void ModifySharedFramework()
+        {
+            // Current working directory is the <coreclr repo root>/sandbox directory.
+            Console.WriteLine($"Modifying the shared framework.");
+
+            var sourcedi = new DirectoryInfo(Directory.GetCurrentDirectory());
+            var targetdi = new DirectoryInfo(
+                new DirectoryInfo(Path.Combine(s_jitBenchDevDirectory, ".dotnet", "shared", "Microsoft.NETCore.App"))
+                .GetDirectories("*")
+                .OrderBy(s => s.Name)
+                .Last()
+                .FullName);
+
+            Console.WriteLine($"  Source : {sourcedi.FullName}");
+            Console.WriteLine($"  Target : {targetdi.FullName}");
+
+            var compiledBinariesOfInterest = new string[] {
```

Comment:
There's some value in having the test here, instead of in CLI; namely, we can test the effects of changes on the runtime and JIT much more quickly, before the changes are ingested by a full-stack build. We've had to do this in a couple of other places (e.g.: optdata creation), so at a minimum I think it would be worthwhile to script this patching separately, and have a single, well-tested, CI-verified script that we could just call to produce a CLI drop out of a runtime build. If we can do something less fragile (e.g., move all such validation to a full-stack build), then I'm all for it.